compass/job/__init__.py

```diff
@@ -0,0 +1,83 @@
+from jinja2 import Template
+from importlib import resources
+import os
+import numpy as np
+
+
+def write_job_script(config, machine, target_cores, min_cores, work_dir):
+    """
+
+    Parameters
+    ----------
+    config : compass.config.CompassConfigParser
+        Configuration options for this test case, a combination of user configs
+        and the defaults for the machine and MPAS core
+
+    machine : {str, None}
+        The name of the machine
+
+    target_cores : int
+        The target number of cores for the job to use
+
+    min_cores : int
+        The minimum number of cores for the job to use
+
+    work_dir : str
+        The work directory where the job script should be written
+    """
+
+    if config.has_option('parallel', 'account'):
+        account = config.get('parallel', 'account')
+    else:
+        account = ''
+
+    cores_per_node = config.getint('parallel', 'cores_per_node')
+
+    # as a rule of thumb, let's do the geometric mean between min and target
+    cores = np.sqrt(target_cores*min_cores)
+    nodes = int(np.ceil(cores/cores_per_node))
+
+    partition = config.get('job', 'partition')
+    if partition == '<<<default>>>':
+        if machine == 'anvil':
+            # choose the partition based on the number of nodes
+            if nodes <= 5:
+                partition = 'acme-small'
+            elif nodes <= 60:
+                partition = 'acme-medium'
+            else:
+                partition = 'acme-large'
+        elif config.has_option('parallel', 'partitions'):
+            # get the first, which is the default
+            partition = config.getlist('parallel', 'partitions')[0]
+        else:
+            partition = ''
+
+    qos = config.get('job', 'qos')
+    if qos == '<<<default>>>':
+        if config.has_option('parallel', 'qos'):
+            # get the first, which is the default
+            qos = config.getlist('parallel', 'qos')[0]
+        else:
+            qos = ''
+
+    constraint = config.get('job', 'constraint')
+    if constraint == '<<<default>>>':
+        if config.has_option('parallel', 'constraints'):
+            # get the first, which is the default
+            constraint = config.getlist('parallel', 'constraints')[0]
+        else:
+            constraint = ''
+
+    job_name = config.get('job', 'job_name')
+    wall_time = config.get('job', 'wall_time')
+
+    template = Template(resources.read_text(
+        'compass.job', 'template.sh'))
+
+    text = template.render(job_name=job_name, account=account,
+                           nodes=f'{nodes}', wall_time=wall_time, qos=qos,
+                           partition=partition, constraint=constraint)
+    script_filename = os.path.join(work_dir, 'job_script.sh')
```

Comment:
How about adding 'compass' in the name?  And then possibly the suite name or test case number (see my other comment)?  E.g. `compass_job_script.suite_full_integration.sh` or `compass_job_script.test_61.sh`. If that's too complicated or requires too many changes, it's fine as it.

Reply:
I agree that having the suite name makes total sense.  I think the suite name will just be `custom` when you set up test cases with numbers.  I don't think `test_61_62_63` or anything like that is very helpful.  I'm not sure if that information is easily available by the time the job script is being created either.

Reply:
> How about adding 'compass' in the name?

I did this but I'm not sure why it's important.  I would think the compass work directory would implicitly only be for doing stuff with compass.